diff --git a/Makefile.in b/Makefile.in
index d087e9e..745e687 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -71,13 +71,10 @@ charset.${O}: compose.uni ubin.uni wide.uni
 
 ${OBJ}: ${srcdir}/less.h ${srcdir}/funcs.h defines.h 
 
-install: all ${srcdir}/less.nro ${srcdir}/lesskey.nro ${srcdir}/lessecho.nro installdirs
+install: all installdirs
 	${INSTALL_PROGRAM} less$(EXEEXT) ${DESTDIR}${bindir}/${binprefix}less$(EXEEXT)
 	${INSTALL_PROGRAM} lesskey$(EXEEXT) ${DESTDIR}${bindir}/${binprefix}lesskey$(EXEEXT)
 	${INSTALL_PROGRAM} lessecho$(EXEEXT) ${DESTDIR}${bindir}/${binprefix}lessecho$(EXEEXT)
-	${INSTALL_DATA} ${srcdir}/less.nro ${DESTDIR}${mandir}/man${manext}/${manprefix}less.${manext}
-	${INSTALL_DATA} ${srcdir}/lesskey.nro ${DESTDIR}${mandir}/man${manext}/${manprefix}lesskey.${manext}
-	${INSTALL_DATA} ${srcdir}/lessecho.nro ${DESTDIR}${mandir}/man${manext}/${manprefix}lessecho.${manext}
 
 install-strip:
 	${MAKE} INSTALL_PROGRAM='${INSTALL_PROGRAM} -s' install
@@ -89,9 +86,6 @@ uninstall:
 	rm -f ${DESTDIR}${bindir}/${binprefix}less$(EXEEXT)
 	rm -f ${DESTDIR}${bindir}/${binprefix}lesskey$(EXEEXT)
 	rm -f ${DESTDIR}${bindir}/${binprefix}lessecho$(EXEEXT)
-	rm -f ${DESTDIR}${mandir}/man${manext}/${manprefix}less.${manext}
-	rm -f ${DESTDIR}${mandir}/man${manext}/${manprefix}lesskey.${manext}
-	rm -f ${DESTDIR}${mandir}/man${manext}/${manprefix}lessecho.${manext}
 
 info:
 install-info:
